downstream.go

```diff
@@ -1448,10 +1454,33 @@ func (dc *downstreamConn) handleMessageRegistered(msg *irc.Message) error {
 				unmarshaledText = dc.unmarshalText(uc, text)
 			}
 			uc.SendMessageLabeled(dc.id, &irc.Message{
+				Tags:    tags,
 				Command: "NOTICE",
 				Params:  []string{upstreamName, unmarshaledText},
 			})
 		}
+	case "TAGMSG":
+		var targetsStr string
+		if err := parseMessageParams(msg, &targetsStr); err != nil {
+			return err
+		}
+		tags := copyClientTags(msg.Tags)
+
+		for _, name := range strings.Split(targetsStr, ",") {
+			uc, upstreamName, err := dc.unmarshalEntity(name)
+			if err != nil {
+				return err
+			}
+			if !uc.caps["message-tags"] {
```

Comment:
This should already be taken care of in `SendMessage`. But maybe we should only have the check in one place?